modules/apps/account/account-test/src/testIntegration/java/com/liferay/account/service/test/AccountEntryUserRelLocalServiceTest.java

```diff
@@ -276,6 +280,98 @@ public void testGetAccountEntryUserRelsByAccountEntryId() throws Exception {
 		Assert.assertArrayEquals(expectedUserIds, actualUserIds);
 	}
 
+	@Test
+	public void testUpdateAccountEntryUserRels() throws Exception {
+		long userId = _user.getUserId();
+
+		// Test add account entries to a user.
+
+		long[] addAccountEntryIds = _addAccountEntries(3);
+		long[] deleteAccountEntryIds = _addAccountEntries(5);
+
+		_accountEntryUserRelLocalService.UpdateAccountEntryUserRels(
+			addAccountEntryIds, deleteAccountEntryIds, userId);
+
+		List<AccountEntryUserRel> accountEntryUserRels =
+			_accountEntryUserRelLocalService.
+				getAccountEntryUserRelsByAccountUserId(userId);
+
+		Assert.assertEquals(
+			accountEntryUserRels.toString(), 3, accountEntryUserRels.size());
+
+		// Test add and delete account entries to a user.
+
+		long[] addAccountEntryIds2 = _addAccountEntries(5);
+		long[] deleteAccountEntryIds2 = {addAccountEntryIds[0]};
+
+		_accountEntryUserRelLocalService.UpdateAccountEntryUserRels(
+			addAccountEntryIds2, deleteAccountEntryIds2, userId);
+
+		List<Long> expectedAccountEntryIdsList = new ArrayList<>(
+			ListUtil.fromArray(addAccountEntryIds2));
+
+		expectedAccountEntryIdsList.add(addAccountEntryIds[1]);
+		expectedAccountEntryIdsList.add(addAccountEntryIds[2]);
+
+		accountEntryUserRels =
+			_accountEntryUserRelLocalService.
+				getAccountEntryUserRelsByAccountUserId(userId);
+
+		List<Long> actualAccountEntryIds = TransformUtil.transform(
+			accountEntryUserRels, AccountEntryUserRelModel::getAccountEntryId);
+
+		Assert.assertEquals(
+			ListUtil.sort(expectedAccountEntryIdsList),
+			ListUtil.sort(actualAccountEntryIds));
+
+		// Test remove all account entries from a user.
+
+		long[] addAccountEntryIds3 = new long[0];
+		long[] deleteAccountEntryIds3 = ArrayUtil.toLongArray(
+			expectedAccountEntryIdsList);
+
+		_accountEntryUserRelLocalService.UpdateAccountEntryUserRels(
+			addAccountEntryIds3, deleteAccountEntryIds3, userId);
+
+		List<AccountEntryUserRel> accountEntryUserRels2 =
+			_accountEntryUserRelLocalService.
+				getAccountEntryUserRelsByAccountUserId(userId);
+
+		Assert.assertEquals(
+			accountEntryUserRels2.toString(), 1, accountEntryUserRels2.size());
+
+		AccountEntryUserRel accountEntryUserRel = accountEntryUserRels2.get(0);
+
+		Assert.assertEquals(
+			AccountConstants.ACCOUNT_ENTRY_ID_DEFAULT,
+			accountEntryUserRel.getAccountEntryId());
+	}
+
+	@Test(expected = DuplicateAccountEntryIdsException.class)
+	public void testUpdateAccountEntryUserRelsThrowsDuplicateAccountEntryIdsException()
+		throws Exception {
+
+		long userId = _user.getUserId();
+
+		long[] addAccountEntryIds = _addAccountEntries(3);
+
+		_accountEntryUserRelLocalService.UpdateAccountEntryUserRels(
+			addAccountEntryIds, addAccountEntryIds, userId);
+	}
+
+	private long[] _addAccountEntries(int count) throws Exception {
+		long[] accountEntryIds = new long[count];
+
+		for (int i = 0; i < count; i++) {
+			AccountEntry accountEntry = AccountEntryTestUtil.addAccountEntry(
```

Comment:
We should track these account entries and make sure to remove them after the tests.